### PR TITLE
Correct paths

### PR DESCRIPTION
**Pull request by eduardogmisiuk**
* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
There are some parts of the code that we do not use absolute paths and do not use the project root as the starting point.

* **What is the new behavior (if this is a feature change)?**
All code related to paths was reformulated to use absolute paths based on the project root.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)

* **Other information**:

### CODE REVIEW

File: lib/configManager/Utils.js [hunk 1]
```diff
@@ -9,14 +11,17 @@ const Types = ['boolean', 'float', 'integer', 'string', 'string[]'];
...
  *
- * @param {string} filename
  * @param {string} path
+ * @param {string} filename
  *
  * @returns {string}
  */
-const createFilename = (filename, path) => `${path}/${filename.toLowerCase()}`;
+const createFilename = (path, filename) => {
+  const rootPath = dirname(require.main.filename);
```
Comment (thiagodpf) [inline lib/configManager/Utils.js:22]: A detail here is that when executing unit tests with Jest, this path (`require.main.filename`) will be, for example: `<project_root>/tests/unitary/some-file.test.js`. That is, ConfigManager will look for a `.conf` file next to my test file.
Ideally, I could pass the root directory (thinking about the testing scenario).

To get around this problem, I had to create a mock through the file `jest.setup.js` that is at the root of the project, so it was possible to establish a correct path to the `.conf` file.

```js
/** This script is at the root of the project! */

/** Mock to fix the problem with the fixed path of the default configuration file. */
require('@dojot/microservice-sdk/lib/configManager/fileManager/Reader').readDefaultConfig = () => {
  const data = readFileSync('./config/default.conf').toString();
  const config = Sanitizer.sanitize(data);
  return config;
};
```
My suggestion is: Give the developer autonomy to inform the root directory (as well as in the method `readUserConfig()`) before using `require.main.filename`. That is, if no root directory path is informed by the developer, use `require.main.filename`.

**Review by thiagodpf — DISMISSED** [review]
I think the code is good, but I didn't notice the difference in relation to how it was before, because the `readDefaultConfig()` method is still with the fixed path and is based only on `require.main.filename`.

**Review by thiagodpf — DISMISSED** [review]
the solution was elegant!

**Comment by thiagodpf** [comment]
DeepScan doesn't seem to understand what an optional parameter is.
If so, use Object destructuring to work around the problem.

**Review by thiagodpf — APPROVED** [review]

**Review by mprevide — APPROVED** [review]

**Pull request merged**